### ISSUE TITLE
Can storeon/router be used with Typescript?

### ISSUE THREAD

**Issue by Hubro** [storeon/router#10]
My line of code to fetch the current route fails with a Typescript error:

```typescript
const {[router.key]: route} = useStoreon<IState, IEvents>(router.key);
```

```
Argument of type 'symbol' is not assignable to parameter of type 'never'.
```

This is because `router.key` is not defined in my state interface (`IState`). I tried adding it:

```typescript
export interface IState {
    [router.key]: any;
}
```

But this fails with this error:

```
A computed property name in an interface must refer to an expression whose type is a
literal type or a 'unique symbol' type. ts(1169)
```

This *should* work, since `router.key` is in fact a symbol literal, but I'm guessing Typescript doesn't know that? The Typescript definitions just say:

```typescript
  export const key: symbol;
```

So can't this router be used with strict Typescript? Is there any way to work around this?

**Comment by Hubro** [comment]
Oh, apparently this is solved by just defining the symbols as `unique symbol`, a type that's available from TypeScript 2.7 and onwards. It lets symbols be used as literals, which lets them be used as computer properties for interfaces.

I made a pull request for the changes: https://github.com/storeon/router/pull/11

**Comment by ivan-solovev** [comment]
I will check the typing and release a new version. Tomorrow.
